### DISCUSSION TITLE
About using vercel.json as reverse proxy

### DISCUSSION THREAD

**Discussion by cennne** [vercel/community#7304] [Feature Question]
### Question

Dear,
I want to ask that can we use vercel.json with this feature: proxy_set_header like nginx ?
I want to set some specific headers, such as Host, X-Forwarded-Host, Key_A(some stuff so I can verify that this request is from vercel)
Thank you

**Comment by versecafe** [comment]
@cennne you can set headers but you'd have to hardcode the host and keys
```json
{
  "headers": [
    {
      "source": "/specific/:path*",
      "headers": [
        {
          "key": "X-Forwarded-Host",
          "value": "host-name"
        }
      ]
    }
  ]
}
```

Or the other hand you can deploy with a `middleware.mjs` and do custom runtime logic for whatever you want

```mjs
import { rewrite } from "@vercel/edge";

export const config = {
  matcher: ["/specific-route-seg/:path*", "/other-route-seg/:path*"],
};

export default function middleware(request) {
  const url = new URL(request.url);

  // rewrite add headers, custom logic, etc
  if (url.pathname.startsWith("/specific-route-seg")) {
    return rewrite(new URL("/specific-route-seg", request.url));
  }
}
```
> https://vercel.com/docs/functions/edge-middleware/middleware-api

**Comment by cennne** [comment]
As I know, the first method is return headers to clients, not send headers to that location. Anyway, thank you

[a comment by an automated account is not shown]
